src/components/ui4/DotsBackground.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react'
+import { LayoutChangeEvent, StyleSheet, View } from 'react-native'
+import { Circle, Defs, G, RadialGradient, Stop, Svg } from 'react-native-svg'
+
+import { useHandler } from '../../hooks/useHandler'
+import { ThemeDot } from '../../types/Theme'
+import { useTheme } from '../services/ThemeContext'
+
+interface Props {
+  accentColor?: string
+}
+
+export function DotsBackground(props: Props): JSX.Element {
+  const { accentColor } = props
+  const theme = useTheme()
+  const { blurR, color, dotOpacity, dots } = theme.background
+
+  const accentDots = [...dots]
+  if (accentColor != null) {
+    accentDots[0] = { ...accentDots[0], color: accentColor }
+  }
+
+  const [{ width, height }, setLayout] = React.useState({ width: 0, height: 0 })
+  const handleLayout = useHandler((event: LayoutChangeEvent) => {
+    setLayout(event.nativeEvent.layout)
+  })
+
+  /**
+   * We are simulating the blur using a radial gradient,
+   * since it's cheaper and looks better.
+   *
+   * To do this, we slice a ring off the outside of each dot and
+   * replace it with the gradient. The sliced-off ring has a width of blurR,
+   * and the gradient ring has a width of 2 * blurR.
+   * This means the circle radius will expand by blurR.
+   *
+   * If the circle is smaller than blurR,
+   * then we can't slice off a big enough ring,
+   * so we reduce the opacity towards 0 to compensate.
+   */
```

Comment:
I have a vague idea on how this works. Looking closely it does look like its the correct implementation for this description, but I have to double check math more to verify.

Reply:
To do a blur mathematically, imagine a circle of radius `blurRadius`, which slides over the scene from left to right. The average color inside the circle becomes the new color for the pixel at the very center of the circle. This imaginary circle is called the "filter kernel" in signal processing lingo.

As it slides, the filter kernel first begins to intersect the background dot at a distance of `blurRadius` away. As it continues to slide through, the ratio of dot color to background color increases, until the filter kernel is fully inside the background dot. This happens when the kernel is `blurRadius` in. So, the total distance from 0% to 100% opacity is `2 * blurRadius`.

The "filter kernel" corresponds to the camera's shutter or iris in physical reality, where a single point on the film receives color from the entire circular lens. If the camera is out-of-focus, this circle will be projected out into space and perform the averaging operation I just described.

Reply:
This seems like something that could be better understood by a video. Though, thanks for clarifying as it does help me better understand blur algorithms.